chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1021 @@
+        "BATR": ["1:3", "1:5", "1:7", "1:9", "1:12"],
+        "VOXL": ["OFF"] + [str(x) for x in range(1, 10)],
+        "VOXD": ["0.5", "1", "1.5", "2", "2.5", "3"],
+    }
+
+    def get_settings(self):
+        main = RadioSettingGroup("Main Settings", "Main")
+        vox = RadioSettingGroup("VOX Settings", "VOX")
+        radio_settings = RadioSettings(main, vox)\
+
+        lists = [
+            ("DCSTM", main, "DCS Tail Mode"),
+            ("TOT", main, "Time out(sec)"),
+            ("VP", main, "Voice Prompts"),
+            ("LCDT", main, "LCD Timeout"),
+            ("BATR", main, "Battery Save"),
+            ("VOXL", vox, "VOX Level"),
+            ("VOXD", vox, "VOX Delay Time(sec)"),
+        ]
+
+        bools = [
+            ("BATS", main, "Battery Save"),
+            ("BEEP", main, "Beep"),
+            ("WARN", main, "Warn"),
+            ("SCAN", main, "Scan"),
+            ("CPYC", main, "Copy Channel"),
+            ("VOXF", vox, "VOX Function"),
+        ]
+
+        ints = [
+            ("SQL", main, "Squelch", 0, 9),
+            ("CH", main, "Channel", 1, 198),
+        ]
+
+        for setting, group, name in bools:
+            value = self._get_bool(setting)
+            rs = RadioSetting(setting, name, RadioSettingValueBoolean(value))
+            group.append(rs)
+
+        for setting, group, name in lists:
+            value = self._get_int(setting)
+            options = self._SETTINGS_OPTIONS[setting]
+            rs = RadioSetting(setting, name,
+                              RadioSettingValueList(options,
+                                                    current_index=value))
+            group.append(rs)
+
+        for setting, group, name, minv, maxv in ints:
+            value = self._get_int(setting)
+            rs = RadioSetting(setting, name,
+                              RadioSettingValueInteger(minv, maxv, value))
+            group.append(rs)
+
+        return radio_settings
+
+    def set_settings(self, settings):
+        for element in settings:
+            if not isinstance(element, RadioSetting):
+                self.set_settings(element)
+                continue
+            if not element.changed():
+                continue
+            elif isinstance(element.value, RadioSettingValueBoolean):
+                self._set_bool(element.get_name(), element.value)
+            elif isinstance(element.value, RadioSettingValueList):
+                # options = self._get_setting_options(element.get_name())
+                options = self._SETTINGS_OPTIONS[element.get_name()]
+                self._set_int(element.get_name(),
+                              options.index(str(element.value)))
+            elif isinstance(element.value, RadioSettingValueInteger):
+                self._set_int(element.get_name(), int(element.value))
+            else:
+                LOG.error("Unknown setting type: %s" % element.value)
```

Comment:
New drivers should ideally be using `MemSetting` which handles symbolic application of almost all the standard types directly to the memory object, eliminating the need for all this bespoke code.